### ISSUE TITLE
RKE1 etcd nodes can not be created with Rancher 2.5.x

### ISSUE THREAD

**Issue by tgeek77** [rancher/rancher#34741]
Rancher installation:
k3s + rancher on VM with 2 cores/4gb ram
Installation is using cert-manager + let's encrypt

RKE cluster installation:
On VMs with 2 cores/4gb ram
VMs using the "rancher-node" security group. Also tried with a custom security group with wide-open permissions as a test.
Both install on existing nodes and install on EC2 nodes affected

Issue: the rancher-agent node never spawns additional kubelet or other containers. In some cases, I can let this run for over an hour with the same issues.

```
root@etcd1:~# docker ps -a
CONTAINER ID   IMAGE                          COMMAND                  CREATED          STATUS          PORTS     NAMES
3f7a40dcb3f6   rancher/rancher-agent:v2.5.9   "run.sh --server htt…"   11 minutes ago   Up 10 minutes             friendly_allen
root@etcd1:~# 
```

Logs from the 
container:
```
root@etcd1:~# docker logs friendly_allen
INFO: Arguments: --server https://rancher.3.68.223.13.nip.io --token REDACTED -r -n m-78pwz
INFO: Environment: CATTLE_ADDRESS=192.168.87.196 CATTLE_AGENT_CONNECT=true CATTLE_INTERNAL_ADDRESS= CATTLE_NODE_NAME=m-78pwz CATTLE_SERVER=https://rancher.3.68.223.13.nip.io CATTLE_TOKEN=REDACTED
INFO: Using resolv.conf: nameserver 127.0.0.53 options edns0 trust-ad search eu-central-1.compute.internal
WARN: Loopback address found in /etc/resolv.conf, please refer to the documentation how to configure your cluster to resolve DNS properly
INFO: https://rancher.3.68.223.13.nip.io/ping is accessible
INFO: rancher.3.68.223.13.nip.io resolves to 3.68.223.13
time="2021-09-15T13:43:27Z" level=info msg="Listening on /tmp/log.sock"
time="2021-09-15T13:43:27Z" level=info msg="Rancher agent version v2.5.9 is starting"
time="2021-09-15T13:43:27Z" level=info msg="Option requestedHostname=m-78pwz"
time="2021-09-15T13:43:27Z" level=info msg="Option customConfig=map[address:192.168.87.196 internalAddress: label:map[] roles:[] taints:[]]"
time="2021-09-15T13:43:27Z" level=info msg="Option etcd=false"
time="2021-09-15T13:43:27Z" level=info msg="Option controlPlane=false"
time="2021-09-15T13:43:27Z" level=info msg="Option worker=false"
time="2021-09-15T13:43:27Z" level=info msg="Connecting to wss://rancher.3.68.223.13.nip.io/v3/connect with token r2mf4dq2cq24gzhdwvpvbpjxcxvx2qqhwf28vldk45hvzzmt59wk4p"
time="2021-09-15T13:43:27Z" level=info msg="Connecting to proxy" url="wss://rancher.3.68.223.13.nip.io/v3/connect"
time="2021-09-15T13:43:27Z" level=info msg="Starting plan monitor, checking every 15 seconds"
```

Occasionally, I will also get this error in addition:

![image](https://user-images.githubusercontent.com/10347457/133453994-54aa1088-2479-4e47-9386-1d7c73e28299.png)

**Comment by sowmyav27** [comment]
Similar to this issue - https://github.com/rancher/rancher/issues/28768

**Comment by tgeek77** [comment]
Update: Changed RKE VMs to t3-large 2core/8gb vms on AWS. 
Result: Same issue

Screenshot:
![image](https://user-images.githubusercontent.com/10347457/133461205-b7e39e66-2d45-43cf-be87-ca791a0e8c86.png)

Logs from container:
```
ubuntu@cplane1:~$ sudo docker ps -a
CONTAINER ID   IMAGE                          COMMAND                  CREATED          STATUS          PORTS     NAMES
aaa02c5c3564   rancher/rancher-agent:v2.5.9   "run.sh --server htt…"   15 minutes ago   Up 15 minutes             admiring_morse
ubuntu@cplane1:~$ sudo docker logs admiring_morse
INFO: Arguments: --server https://rancher.3.68.223.13.nip.io --token REDACTED -r -n m-xvtfp
INFO: Environment: CATTLE_ADDRESS=172.31.31.26 CATTLE_AGENT_CONNECT=true CATTLE_INTERNAL_ADDRESS= CATTLE_NODE_NAME=m-xvtfp CATTLE_SERVER=https://rancher.3.68.223.13.nip.io CATTLE_TOKEN=REDACTED
INFO: Using resolv.conf: nameserver 127.0.0.53 options edns0 trust-ad search eu-central-1.compute.internal
WARN: Loopback address found in /etc/resolv.conf, please refer to the documentation how to configure your cluster to resolve DNS properly
INFO: https://rancher.3.68.223.13.nip.io/ping is accessible
INFO: rancher.3.68.223.13.nip.io resolves to 3.68.223.13
time="2021-09-15T14:58:52Z" level=info msg="Listening on /tmp/log.sock"
time="2021-09-15T14:58:52Z" level=info msg="Rancher agent version v2.5.9 is starting"
time="2021-09-15T14:58:52Z" level=info msg="Option customConfig=map[address:172.31.31.26 internalAddress: label:map[] roles:[] taints:[]]"
time="2021-09-15T14:58:52Z" level=info msg="Option etcd=false"
time="2021-09-15T14:58:52Z" level=info msg="Option controlPlane=false"
time="2021-09-15T14:58:52Z" level=info msg="Option worker=false"
time="2021-09-15T14:58:52Z" level=info msg="Option requestedHostname=m-xvtfp"
time="2021-09-15T14:58:52Z" level=info msg="Connecting to wss://rancher.3.68.223.13.nip.io/v3/connect with token t6vjckxwbg4bdwc5xkcwdp6zxhbljrt5ps7hs7r2xdkppf7svmrt4r"
time="2021-09-15T14:58:52Z" level=info msg="Connecting to proxy" url="wss://rancher.3.68.223.13.nip.io/v3/connect"
time="2021-09-15T14:58:52Z" level=warning msg="Error while getting agent config: invalid response 500: Operation cannot be fulfilled on nodes.management.cattle.io \"m-xvtfp\": the object has been modified; please apply your changes to the latest version and try again"
time="2021-09-15T14:58:57Z" level=info msg="Starting plan monitor, checking every 15 seconds"
ubuntu@cplane1:~$ 
```

**Comment by tgeek77** [comment]
Update: Changed RKE VMs to t3-2xlarge 8 cores/32Gb. 
Result: Same issue

This is not an issue with nodes not having enough RAM/Cores
